staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions.go

```diff
@@ -364,12 +366,17 @@ func (o *EditOptions) Run() error {
 			containsError = false
 			updatedVisitor := resource.InfoListVisitor(updatedInfos)
 
-			// we need to add back managedFields to both updated and original object
-			if err := o.restoreManagedFields(updatedInfos); err != nil {
-				return preservedFile(err, file, o.ErrOut)
-			}
-			if err := o.restoreManagedFields(infos); err != nil {
-				return preservedFile(err, file, o.ErrOut)
+			// add warning about editing managed fields if flag --show-managed-fields is set
+			// Otherwise, add back managedFields to both updated and original object
+			if o.PrintFlags.JSONYamlPrintFlags.ShowManagedFields {
+				fmt.Fprintln(o.ErrOut, "Editing managed fields does not use Server-Side Apply")
```

Comment:
Thanks for the PR. As far as I understand, there is no distinction with client side apply vs server side apply in this case. I think, you need to compare previous manage fields and current manage fields, after that if there is any difference, just discard these from the patch and warn user. Am I missing something @apelisse ?

Reply:
Yes, there's a misunderstanding.

The problem is not that "editing managed fields does not use ssa", the problem is that we don't want people to edit managed fields. We can show it to them, but we don't want them to edit it. So as @ardaguclu mentions above, we want to:
1. Diff managed fields before and after edit
2. If they are different, print a warning (including I think, if they have been entirely removed)
3. Always make sure we don't send them as part of the patch

To be honest, I would always check if old managed fields != new managed fields, even if they had been trimmed (because show-managed-fields == false). For example, if I do `kubectl edit --show-managed-fields=false ...` and then try to add them in the edit, we should still discard them and print a warning.